src/lib/http/http_server.pl

```diff
@@ -335,3 +325,15 @@
             pipe_bytes(StreamIn, StreamOut)
         )
     ;   true).
+
+% WARNING: This only works for ASCII chars. This code can be modified to support
+% Latin1 characters also but a completely different approach is needed for other
+% languages. Since HTTP internals are ASCII, this is fine for this usecase.
+chars_lower(Chars, Lower) :-
+    maplist(char_lower, Chars, Lower).
+char_lower(Char, Lower) :-
```

Comment:
For later, this functionality would be a useful and very welcome extension to `char_type/2` from `library(charsio)`, for example as:

<pre>
<b>?- char_type(X, to_lower('A')).</b>
X = a.
</pre>

Using appropriate Rust features for reasoning about Unicode, this could likely be made efficient and also general.